### PR TITLE
Write correct THRESHPR values also from restart

### PR DESCRIPTION
**Pull request by vkip**
Managed to leave this out of https://github.com/OPM/opm-simulators/pull/5743, sorry about that.

### CODE REVIEW

**Comment by vkip** [comment]
jenkins build this please

File: opm/simulators/flow/GenericThresholdPressure_impl.hpp [hunk 1]
```diff
@@ -227,6 +227,12 @@ getRestartVector() const
         return {};
 
     std::vector<Scalar> result(numEquilRegions_ * numEquilRegions_, 0.0);
+
+    if (this->restart_) {
```
Comment (atgeirr) [inline opm/simulators/flow/GenericThresholdPressure_impl.hpp:231]: This certainly looks correct to me, even to the point where always returning `this->thpres_ ` should be correct, also for non-restart cases? After all, that array is initialized by code identical to the block below?

Comment (vkip) [reply opm/simulators/flow/GenericThresholdPressure_impl.hpp:231]: That makes sense! Updating..

**Comment by vkip** [comment]
jenkins build this please

**Comment by vkip** [comment]
jenkins build this please

File: opm/simulators/flow/GenericThresholdPressure_impl.hpp (hunk 1 above)
Comment (vkip) [reply opm/simulators/flow/GenericThresholdPressure_impl.hpp]: Needed a small fix for parallel without restart, but seems to work.

**Comment by atgeirr** [comment]
Thanks, merging.

**Pull request merged**